cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/RemoteTaskExecutor.java

```diff
@@ -141,9 +142,15 @@ public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws Exception
         }
       }, retryStrategy, Retries.DEFAULT_PREDICATE);
     } catch (Exception e) {
+      Exception ex = e;
+      if (e instanceof ServiceException) {
```

Comment:
Could you use two catch statements? It will have duplicated `emitMetrics` line, but it's less confusing than `e`/`ex` reassignment?

Reply:
done